src/pages/data-entry/data-entry.jsx

```diff
@@ -1,127 +1,214 @@
-import React, { Component } from "react";
+import React, { Component } from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import { useMutation } from '@apollo/react-hooks'
+import { gql } from 'apollo-boost'
+
+const units = gql`
+  {
+    getAllAvailableUnits
+    }
+  }
+`
+
+const insertMutation = gql`
+  mutation InsertMutation(
+    $unit: String!
+    $value: Float!
+    $latitude: Float!
+    $longitude: Float!
+  ) {
+    insertValue(
+      unit: $unit
+      value: $value
+      latitude: $latitude
+      longitude: $longitude
+    )
+  }
+`
+
+function DataFunction () {
+  const { loading, error, data } = useQuery(units)
+
+  if (loading) return 'Loading...'
+  if (error) return `Error! ${error.message}`
+
+  return data.getAllAvailableUnits
+}
 
 export class DataEntryForm extends Component {
-  constructor(props) {
-    super(props);
+  constructor (props) {
+    super(props)
 
     this.state = {
-      age: "",
-      sex: "",
-      weight: "",
-      wunits: "",
-      height: "",
-      hunits: "",
+      age: '',

```

Comment:
Shouldn't this be a number?

Reply:
Reason I don't have it as a number is that initializing it to a zero or something gave errors in the webpage. Can explain that more in person.